src/components/Sheet/Sheet.tsx

```diff
@@ -39,6 +39,10 @@ export interface Props {
   children: React.ReactNode;
   /** Callback when the backdrop is clicked or `ESC` is pressed */
   onClose(): void;
+  /** Callback when the sheet has completed entering */
+  onEntered?(): void;
+  /** Callback when the sheet has started to exit */
+  onExit?(): void;
```

Comment:
Added this specifically to solve the issue where the sheet transition would skip directly to the end when I child element had focus.

|Before|After|
|---|---|
|![before](https://user-images.githubusercontent.com/344839/59897113-02a61500-93a0-11e9-9329-4ef2dd6a808e.gif)|![after](https://user-images.githubusercontent.com/344839/59897120-0cc81380-93a0-11e9-8302-13a4acb2f304.gif)|

Although, there still appears to be an issue with the range slider:

![range slider](https://user-images.githubusercontent.com/344839/59897307-e9ea2f00-93a0-11e9-9401-1acdb4bb528d.gif)

Reply:
@danrosenthal did some digging and looks like any children which cause a reflow during the entering animation will force the animation to skip to the end. https://github.com/reactjs/react-transition-group/issues/382